example_test.go

```diff
@@ -0,0 +1,46 @@
+package quamina_test
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/timbray/quamina"
+)
+
+const userRegisteredEvent = `{
+  "id": "1c0e1ce4-3d88-4786-a09d-7133c170d02a",
+  "type": "UserRegistered",
+  "user": {
+    "name": "Doe, John",
+    "premiumAccount": true
+  }
+}
+`
+
+const premiumUserPattern = `{
+  "type":["UserRegistered"],
+  "user": {"premiumAccount": [true]}
+}`
+
+func ExampleNew() {
+	q, err := quamina.New()
+	if err != nil {
+		log.Fatalf("could not create quamina instance: %v", err)
+	}
+
+	err = q.AddPattern("premium user", premiumUserPattern)
+	if err != nil {
+		log.Fatalf("could not add pattern: %v", err)
+	}
+
+	event, err := q.MatchesForEvent([]byte(userRegisteredEvent))
```

Comment:
not crazy about the variable name "event" - what you get back is []X, so a better name would be "matches" or something like that.

Reply:
Ah, of course! Will do.